### DISCUSSION TITLE
Aggiornamento Hassos

### DISCUSSION THREAD

**Discussion by angwqe** [indomus/forum#1200] [HUB: Home Assistant]
Scusate ma ultimamente ho fatto un aggiornamento di asso, l'ultimo uscito e da allora non mi funziona più l'integrazione di Amazon Alexa (quella che da problemi continui da sempre) 

mi dice che non l'integrazione non è caricata. è integrata correttamente perché funzionava ed anche tutti i dispositivi risultano visibili, ma non posso più controllarla

cosa può essere successo?

**Comment by indomus** [comment]
Se ti riferisci ad Alexa Media Player, controlla su HACS che non ci siano aggiornamenti

**Reply by angwqe** [reply]
si mi riferisco ad Alexa media player, ora controllo ma mi risultava aggiornato

**Comment by Dregi56** [comment]
Non mi ero accorto del tuo post, anche perchè l'oggetto della discussione è diverso.Ho lo stesso problema. HA per l'integrazione Alexa media Player mi scriveva: non caricato. Ho pensato di reinstallare l'integrazione, ma all'installazione mi dice: **Impossibile caricare il flusso di configurazione**. Sono bloccato. Non so che fare.

**Reply by Scialla** [reply]
C'è un aggiornamento (dell'integrazione) su HACS che sistema

**Reply by Dregi56** [reply]
Ho provato ad installare HACS, ed è andata  a buon fine la registrazione device su GitHub, almeno GitHub ha risposto positivamente all'autorizzazione. anche in HA la finestra pop-up ha detto configurazione riuscita. Solo che, come succede per Alexa media Player sulla card dell'integrazione c'è scritto **non caricato** e **configurazione non riuscita** con l'invito a controllare il registro. Il nuovo errore nel Log:

Questo errore ha avuto origine da un'integrazione personalizzata.

Logger: homeassistant.config_entries
Source: custom_components/hacs/tasks/setup_frontend.py:6
Integration: HACS (documentation, issues)
First occurred: 15:46:30 (1 occurrences)
Last logged: 15:46:30

Error setting up entry for hacs
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 313, in async_setup
    result = await component.async_setup_entry(hass, self)  # type: ignore
  File "/config/custom_components/hacs/__init__.py", line 39, in async_setup_entry
    return await hacs_ui_setup(hass, config_entry)
  File "/config/custom_components/hacs/operational/setup.py", line 94, in async_setup_entry
    await _async_common_setup(hass)
  File "/config/custom_components/hacs/operational/setup.py", line 50, in _async_common_setup
    await hacs.tasks.async_load()
  File "/config/custom_components/hacs/tasks/manager.py", line 43, in async_load
    await asyncio.gather(*[_load_module(task) for task in task_modules])
  File "/config/custom_components/hacs/tasks/manager.py", line 39, in _load_module
    task_module = import_module(f"{__package__}.{module}")
  File "/usr/local/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/config/custom_components/hacs/tasks/setup_frontend.py", line 6, in <module>
    from ..hacs_frontend import locate_dir
ModuleNotFoundError: No module named 'custom_components.hacs.hacs_frontend'

Non sono molto esperto di voci di registro. Cosa devo fare?

**Reply by Dregi56** [reply]
Anche se sono interessato ad avere HACS, in che modo potrei installare egualmente l'aggiornamento ad Alexa_Media_Player per ovviare al bug relativo a questa integrazione?

**Reply by Scialla** [reply]
io ripartirei... rimuoverei sia alexa che HACS e ricomincerei a reinstallare HACS seguendo pedessiquamente la guida. Come vedi dal log, non risulta installato..
Inoltre HACS (a memoria) non funziona immediatamente appena installato ma ci và qualche ora....

**Reply by Dregi56** [reply]
Ok. ci provo, anche se hassio è meno duttile delle altre soluzioni di SO.

**Reply by Dregi56** [reply]
Scialla, ci son riuscito, anche se ho "scoperto" delle cose(a)....  Comunque HACS funziona e mi ha permesso di risolvere come da te indicato AlexaMediaPlayer che ora è a posto. Si HACS funziona, ma tra le integrazioni la card relativa recita: **Configurazione non riuscita** con un "bel" punto esclamativo rosso! Anche se, ripeto, funziona.
(a) Sono riuscito ad installarlo senza che facesse errori solo "commentando" la riga del token in configuration.yaml, poi un riavvio e quindi ripristinando la riga.
